File: app/internal/publisher/template/java/widgets/DashboardWidget.java
package com.websmith.sdk.widgets;

import com.websmith.sdk.LicenseEngine;
import com.websmith.sdk.LicenseEngine.LicenseStatus;

public class DashboardWidget {
    public static void render() {
        System.out.println("\n=== License Dashboard ===\n");
        try {
            LicenseEngine engine = new LicenseEngine();
            LicenseStatus status = engine.initialize();
            System.out.println("License Status: " + status.status);
            System.out.println("Valid: " + (status.valid ? "Yes" : "No"));
            if (status.expiresAt != null) {
                System.out.println("Expires: " + status.expiresAt);
            }
            System.out.println("Days Remaining: " + status.daysRemaining);
            if (status.plan != null) {
                System.out.println("Plan: " + status.plan);
            }
            if (status.hardwareId != null) {
                System.out.println("Hardware ID: " + status.hardwareId);
            }
            if (status.licenseKey != null) {
                System.out.println("License Key: " + status.licenseKey);
            }
            if (status.message != null) {
                System.out.println("Message: " + status.message);
            }
            System.out.println("Trial Active: " + (status.trialActive ? "Yes" : "No"));
            String line = "----------------------------------------";
            System.out.println("\n" + line);
            System.out.println("License Key Available: " + (engine.hasLicenseKey() ? "Yes" : "No"));
            System.out.println("Hardware ID: " + engine.getHardwareId());
        } catch (Exception e) {
            System.out.println("Dashboard error: " + e.getMessage());
        }
    }
}

File: app/internal/publisher/template/php/widgets/dashboard_widget.php
<?php
namespace WSD\SDK\Widgets;

use WSD\SDK\LicenseEngine;

class DashboardWidget
{
    public static function render(): void
    {
        echo "\n=== License Dashboard ===\n\n";
        try {
            $engine = new LicenseEngine();
            $status = $engine->initialize();
            echo "License Status: {$status->status}\n";
            echo "Valid: " . ($status->valid ? 'Yes' : 'No') . "\n";
            if ($status->expiresAt !== null) {
                echo "Expires: {$status->expiresAt}\n";
            }
            echo "Days Remaining: {$status->daysRemaining}\n";
            if ($status->plan !== null) {
                echo "Plan: {$status->plan}\n";
            }
            if ($status->hardwareId !== null) {
                echo "Hardware ID: {$status->hardwareId}\n";
            }
            if ($status->licenseKey !== null) {
                echo "License Key: {$status->licenseKey}\n";
            }
            if ($status->message !== null) {
                echo "Message: {$status->message}\n";
            }
            echo "Trial Active: " . ($status->trialActive ? 'Yes' : 'No') . "\n";
            echo "\n---\n";
            if ($engine->hasLicenseKey()) {
                echo "License Key Available: Yes\n";
                echo "Hardware ID: {$engine->getHardwareId()}\n";
            } else {
                echo "License Key Available: No\n";
            }
        } catch (\Throwable $e) {
            echo "Dashboard error: {$e->getMessage()}\n";
        }
    }
}
